docs/DeveloperGuide.md

```diff
@@ -389,144 +363,284 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
-2. PlaNus shows a list of tasks.
+2. User requests to find tasks with specified search phrase(s) in specified attribute(s).
+
+3. PlaNus shows a list of tasks that match the specified search phrase(s) in the specified attribute(s).
 
   Use case ends.
 
 **Extensions**
 
-  - 2a. The list is empty.
+- 2a. The search phrase is empty or consists of only white spaces.
+
+  - 2a1. PlaNus shows an error message.
+
+    Use case ends.
+
+- 2b. The search phrase is in invalid format or includes invalid characters.
+
+    - 2b1. PlaNus shows an error message.
 
     Use case ends.
 
+- 3a. The list is empty.
+
+  Use case ends.
+
 <br>
 
-**Use case 07: Request help**
+**Use case: UC06 - Edit a task**
+
+**Guarantees**
+
+* A task will be edited only if the task is still valid after being edited.
 
 **MSS**
 
-1. User requests help.
+1. User <ins>lists a set of tasks (UC03 or UC05)</ins>
 
-2. PlaNus shows available commands.
+2. User requests to edit values of specified attribute(s) of a task.
+
+3. PlaNus shows task with updated attribute(s).
 
   Use case ends.
 
+**Extensions**
+- 2a. The given index is invalid.
+
+  - 2a1. PlaNus shows an error message.
+
+    Use case resumes at step 1.
+
+- 2b. The given value of an attribute is invalid.
+
+  - 2b1. PlaNus shows an error message.
+
+    Use case resumes at step 1.
+
 <br>
 
-**Use case 08: Mark a task as done**
+**Use case: UC07 - Delete tasks**
 
 **MSS**
 
-1. User requests to list tasks.
-
-2. PlaNus shows a list of tasks.
+1. User <ins>lists a set of tasks (UC03 or UC05)</ins>
 
-3. User requests to mark a specific tasks in the list as done.
+2. User requests to delete certain tasks from the list.
 
-4. PlaNus marks the task as done.
+3. PlaNus deletes the tasks.
 
   Use case ends.
 
 **Extensions**
 
-- 3a. The given index is invalid.
+- 1a. The list is empty.
+
+  Use case ends.
+
+- 2a. Some indexes given are invalid.
 
   - 3a1. PlaNus shows an error message.
 
     Use case resumes at step 3.
 
 <br>
 
-**Use case 09: Find a task**
+**Use case: UC08 - Add a lesson**
+
+**Guarantees**
+
+* A new lesson will be added only there is no duplicate lesson already existing in PlaNus.
 
 **MSS**
 
-1. User requests to find task with specified search phrase(s) in specified attribute(s).
+1. User requests to add a lesson to the list.
 
-2. PlaNus shows a list of tasks that match the specified search phrase(s) in the specified attribute(s).
+2. PlaNus adds the lessons as recurring tasks.
 
   Use case ends.
 
+
 **Extensions**
 
-- 1a. The search phrase is empty or consists of only white spaces.
+- 1a. The given input is invalid.
 
   - 1a1. PlaNus shows an error message.
 
+    Use case resumes at step 1.
+
+- 1b. The lesson already exists in the list.
+
+  - 1b1. PlaNus shows an error message.
+
+    Use case resumes at step 1.
+
+<br>
+
+**Use case: UC09 - Find lessons**
+
+**MSS**
+
+1. User <ins>lists a set of lessons (UC08 or UC09)</ins>
```

Comment:
Why is it UC08/09

Reply:
Same. find can also list a set of lessons